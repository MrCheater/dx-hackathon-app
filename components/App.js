import React from 'react'
import PropTypes from 'prop-types'

const App = ({children}) => (
  <div className="root">
    {children}
  </div>
)

App.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ])
}

export default App
