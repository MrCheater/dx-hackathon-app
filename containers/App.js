import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

export class App extends React.PureComponent {
  componentDidMount() {
    if(!this.props.isLogged) {
      window.location.replace('/auth')
    }
  }

  render() {
    return (
      <div className="root">
        {this.props.isLogged ? this.props.children : null}
      </div>
    )
  }
}

App.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ])
}

export const mapStateToProps = (state) => ({
  isLogged: !!state.user.upn
})

export default connect(mapStateToProps)(App)
