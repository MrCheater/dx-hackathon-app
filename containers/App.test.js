import React from 'react'
import { App } from './App'
import { shallow } from 'enzyme'

test('App must match snapshot', () => {
  const markup = shallow(
    <App isLogged={true}>
      Test
    </App>
  )
  expect(markup).toMatchSnapshot()
})
