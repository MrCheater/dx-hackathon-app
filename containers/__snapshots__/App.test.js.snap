// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App must match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="root"
>
    Test
</div>,
  "nodes": Array [
    <div
      className="root"
>
      Test
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <App
        isLogged={true}
>
        Test
</App>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": "Test",
          "isLogged": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="root"
>
          Test
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="root"
>
          Test
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App
    isLogged={true}
>
    Test
</App>,
}
`;
